src/DbInterface.cpp

```diff
@@ -513,9 +513,8 @@ void DbInterface::handlePostLinkProberMetrics(
         mLinkProbeMetrics[static_cast<int> (metrics)]
     );
 
-    if (metrics == link_manager::ActiveStandbyStateMachine::LinkProberMetrics::LinkProberUnknownStart) {
-        mStateDbLinkProbeStatsTablePtr->hdel(portName, mLinkProbeMetrics[0]);
-        mStateDbLinkProbeStatsTablePtr->hdel(portName, mLinkProbeMetrics[1]);
+    for (auto linkProberMetricsEvent:mLinkProbeMetrics) {
+        mStateDbLinkProbeStatsTablePtr->hdel(portName, linkProberMetricsEvent);
     }
```

Comment:
Should we flush the queue when deleting the DB interface object?

Reply:
Hi @yxieca - can you be more specific, which queue should we flush? 

Also here we are only deleting stale timestamps (<field, value> pair) in the table instead of the whole object.